bh_modules/pascalkeywords.py

```diff
@@ -0,0 +1,14 @@
+"""
+BracketHighlighter.
+
+Copyright (c) 2013 - 2015 Isaac Muse <isaacmuse@gmail.com>
+License: MIT
+"""
+
+
+def compare(name, first, second, bfr):
+    """Differentiate 'repeat..until' and 'unit..end.' from '*..end' brackets."""
+    brackets = (bfr[first.begin:first.end], bfr[second.begin:second.end])
+    return (brackets[0] == "repeat" and brackets[1] == "until") \
+        or (brackets[0] == "unit" and brackets[1] == "end.") \
```

Comment:
> I debated adding unit, so I could remove the cases for unit and end.since almost every Pascal file will start with one and end with the other, so highlighting them is almost pointless. Then I could use the simplified logic of lua (repeat XOR end).

I'll leave that up to you.  Just let me know what your final decision is and I think we're ready to go.  Everything seems to look fine.  I am indifferent to whether `unit` stays in or not.  I don't really use Pascal, so I leave those kinds of decisions up to the users of the language.